SearchParser.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * SearchParser
+ * @param {Array} patterns  Enumerates params and the patterns that match them
+ * @param {boolean} options.returnAllMatches  If true return all matches (not just first)
```

Comment:
Is `returnAllMatches` for convenience or performance? If it's for former, I'd say that should be the job of the calling function. If it's the later, I'm not completely sold. These regexes will almost always be fairly simple, as will the search text.